content/docs/security/ratelimit/global.md

```diff
@@ -0,0 +1,432 @@
+
+```yaml
+kubectl apply -f - <<EOF
+apiVersion: gateway.kgateway.dev/v1alpha1
+kind: TrafficPolicy
+metadata:
+  name: ip-rate-limit
+  namespace: {{< reuse "docs/snippets/namespace.md" >}}
+spec:
+  targetRefs:
+  - group: gateway.networking.k8s.io
+    kind: Gateway
+    name: http
+  rateLimit:
+    global:
+      descriptors:
+      - entries:
+        - type: RemoteAddress
+      extensionRef:
+        name: global-ratelimit
+EOF
+```
+
+{{% /tab %}}
+
+{{% tab tabName="Request path" %}}
+
+Limit requests based on the request path, such as a request with `/path1` or `/path2` in the path as defined in the example Rate Limit Service actions.
+
+```yaml
+kubectl apply -f - <<EOF
+apiVersion: gateway.kgateway.dev/v1alpha1
+kind: TrafficPolicy
+metadata:
+  name: request-path
+  namespace: {{< reuse "docs/snippets/namespace.md" >}}
+spec:
+  targetRefs:
+  - group: gateway.networking.k8s.io
+    kind: Gateway
+    name: http
+  rateLimit:
+    global:
+      descriptors:
+      - entries:
+        - type: Path
+      extensionRef:
+        name: global-ratelimit
+EOF
+```
+
+{{% /tab %}}
+
+{{% tab tabName="User ID" %}}
+
+Limit requests based on a user ID from an `X-User-ID` header.
+
+```yaml
+kubectl apply -f - <<EOF
+apiVersion: gateway.kgateway.dev/v1alpha1
+kind: TrafficPolicy
+metadata:
+  name: user-rate-limit
+  namespace: {{< reuse "docs/snippets/namespace.md" >}}
+spec:
+  targetRefs:
+  - group: gateway.networking.k8s.io
+    kind: Gateway
+    name: http
+  rateLimit:
+    global:
+      descriptors:
+      - entries:
+        - type: Header
+          header: "X-User-ID"
+      extensionRef:
+        name: global-ratelimit
+EOF
+```
+
+{{% /tab %}}
+
+{{% tab tabName="Combined rate limiting" %}}
+
+Apply different limits based on both path and user ID:
+
+```yaml
+kubectl apply -f - <<EOF
+apiVersion: gateway.kgateway.dev/v1alpha1
+kind: TrafficPolicy
+metadata:
+  name: combined-rate-limit
+  namespace: {{< reuse "docs/snippets/namespace.md" >}}
+spec:
+  targetRefs:
+  - group: gateway.networking.k8s.io
+    kind: Gateway
+    name: http
+  rateLimit:
+    global:
+      descriptors:
+      - entries:
+        - type: Path
+        - type: Header
+          header: "user-id"
+      extensionRef:
+        name: global-ratelimit
+EOF
+```
+
+{{% /tab %}}
+
+{{% tab tabName="Combined local and global rate limiting" %}}
+
+Combine local and global rate limiting in the same TrafficPolicy. For more information, see [Local rate limiting](../local).
+
+```yaml
+kubectl apply -f - <<EOF
+apiVersion: gateway.kgateway.dev/v1alpha1
+kind: TrafficPolicy
+metadata:
+  name: local-global-rate-limit
+  namespace: {{< reuse "docs/snippets/namespace.md" >}}
+spec:
+  targetRefs:
+  - group: gateway.networking.k8s.io
+    kind: Gateway
+    name: http
+  rateLimit:
+    local:
+      tokenBucket:
+        maxTokens: 5
+        tokensPerFill: 1
+        fillInterval: "1s"
+    global:
+      descriptors:
+      - entries:
+        - type: Generic
```

Comment:
the type here vs. the descriptor entries earlier are kind of confusing. earlier, type matched the descriptor in the RLS (I think), but now this seems to be your own descriptor or something?? and you refer to the descriptor under generic.key

Reply:
yeah, it is confusing...I think that is just how it works for if you want to rate limit based on a key-value descriptor pair that you define vs the built-in path, remote address, or header descriptors. I will try to revise the description a bit.